### PR TITLE
Add ransomware data to CyHy VS report

### LINKED ISSUE #102: Add REV Content to KEV Page in Weekly Report
# 💡 Summary #

Add verbiage to cover the newly added REV content to the weekly VS along with graphics capturing the REV counts.

## Motivation and context ##

The stakeholder needs to know the number of REVs identified from the total number of KEVs.

## Implementation notes ##

**Add the following verbiage to the bottom of the "KEV" page of weekly reports** --This goes for both federal and non-federal

"Signed into law in March 2022, the [Cyber Incident Reporting for Critical Infrastructure Act of 2022 (CIRCIA)](https://www.cisa.gov/topics/cyber-threats-and-advisories/information-sharing/cyber-incident-reporting-critical-infrastructure-act-2022-circia) required CISA to establish the [Ransomware Vulnerability Warning Pilot (RVWP)](https://www.cisa.gov/stopransomware/Ransomware-Vulnerability-Warning-Pilot). The goal of the RVWP is to warn organizations about exposed vulnerabilities that may be exploited by ransomware threat actors."

Of the ___ KEV findings detected on your [organization's/agency's] internet-facing assets, ___ are known by CISA to have been used in ransomware campaigns. We recommend remediating these as soon as possible.

image

Link for CIRCIA Hyperlink: https://www.cisa.gov/topics/cyber-threats-and-advisories/information-sharing/cyber-incident-reporting-critical-infrastructure-act-2022-circia

Link for RVWP Hyperlink: https://www.cisa.gov/stopransomware/Ransomware-Vulnerability-Warning-Pilot

**Add bubbles below the new verbiage for the current REV counts by severity (See above image)**

## Acceptance criteria ##

- [x] New Verbiage added at the bottom of the KEV page
- [x] New Graphic capturing the REV count by severity

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR makes the new KEV ransomware flag available throughout the CyHy Vulnerability Scan (VS) report:
- Adds a new ransomware section to the bottom of the BOD 22-01 (KEV) page (https://github.com/cisagov/cyhy-system/issues/102)
- Adds the `known_ransomware` flag to the `findings.csv` and `recently-detected.csv` attachments (part of https://github.com/cisagov/cyhy-system/issues/101)

In the course of testing these changes, I discovered and fixed an unreported bug that was preventing KEV data from displaying in the `recently-detected.csv` attachment.  The fix can be found in https://github.com/cisagov/cyhy-reports/pull/97/commits/f558a086dc0a30e435f661dcdfcba3b886bd322d.  For this reason, I added the "bug" label to this PR.

## 💭 Motivation and context ##

Now that known-exploited ransomware data is available in the [CISA KEV Catalog](https://www.cisa.gov/known-exploited-vulnerabilities-catalog), we want to make it visible in the CyHy report.

This PR resolves https://github.com/cisagov/cyhy-system/issues/102 as well as part of https://github.com/cisagov/cyhy-system/issues/101.

## 🧪 Testing ##

I built a local Docker image with these changes and then used that to generate a variety of test reports.  I ensured that the reports looked as expected (for both Federal Executive and non-Federal Executive organizations), in terms of the new content changes and modifications to the `findings.csv` and `recently-detected.csv` attachments.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [x] Coordinate with the CyHy team and deploy these changes to Production.

### CODE REVIEW

File: cyhy_report/customer/graphs.py [hunk 1]
```diff
@@ -1224,9 +1224,90 @@ def plot(self, filename, size=1.0):
...
+        [5, 5, 5, 5],  # Make all bubbles the same size
+        BUBBLE_COLORS,
+        ["CRITICAL", "HIGH", "MEDIUM", "LOW"],
+        [7, 89, 305, 12345],
+    )
+    hbc.plot("horizontal-bubbles")
+
     # bc = MyBubbleChart(
     #     # CRIT,HIGH,MED,LOW
     #     [50, 20, 65, 35],  # horizontal location
```
Comment (jsf9k) [inline cyhy_report/customer/graphs.py:1313]: Can we get rid of all this commented-out code?

**Review by jsf9k — APPROVED** [review]
This looks good to me.  I made one suggestion, which you can take or leave.

File: cyhy_report/customer/graphs.py (hunk 1 above)
Comment (dav3r) [reply cyhy_report/customer/graphs.py:1313]: I've left it in on purpose over the years, as it is a helpful refresher whenever I need to work on any of those graphs.  Usually we go a while in between looking at this code, so I find it nice to remind myself that we have an easy way to test out changes in that file.

Comment (jsf9k) [reply cyhy_report/customer/graphs.py:1313]: So be it.

**Review by jasonodoom — APPROVED** [review]

**Pull request merged**